### PR TITLE
Drop use of Distutils LooseVersion

### PR DESCRIPTION
**Pull request by jenshnielsen**
All of distutils is deprecated in python 3.10 and the recommended alternative for LooseVersion is packaging.version.parse

This removes all use of distutils outside versioneer

### CODE REVIEW

File: qcodes/instrument_drivers/Keysight/private/Keysight_344xxA_submodules.py [hunk 1]
```diff
@@ -464,8 +465,7 @@ def __init__(self, name: str, address: str, silent: bool = False,
 
         options = self._options()
         self.has_DIG = self.is_34465A_34470A and (
-            'DIG' in options
-            or LooseVersion('A.03') <= LooseVersion(idn['firmware'])
+            "DIG" in options or version.parse("A.03") <= version.parse(idn["firmware"])
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/Keysight/private/Keysight_344xxA_submodules.py:468]: Looks like versions of this form are considdered legacy and also deprecated using packaging. We should look for an alternative way of doing version comparisons in these cases

File: qcodes/instrument_drivers/Keysight/keysight_e4980a.py [hunk 2]
```diff
@@ -194,9 +194,9 @@ def __init__(self,
 
         idn = self.IDN.get()
 
-        self.has_firmware_a_02_10_or_above = (
-                LooseVersion(idn["firmware"]) >= LooseVersion("A.02.10")
-        )
+        self.has_firmware_a_02_10_or_above = version.parse(
+            idn["firmware"]
+        ) >= version.parse("A.02.10")
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/Keysight/keysight_e4980a.py:199]: same here

File: qcodes/instrument_drivers/Keysight/private/Keysight_344xxA_submodules.py (hunk 1 above)
Comment (jenshnielsen) [reply qcodes/instrument_drivers/Keysight/private/Keysight_344xxA_submodules.py:468]: Perhaps one could simply convert letters to their ascii mapping before passing to parse

```
In [39]: verstr = "a.1.0"
    ...: b = []
    ...: for v in verstr:
    ...:     if v.isalpha():
    ...:         b.append(str(ord(v.upper())))
    ...:     else:
    ...:         b.append(v)
    ...: c = "".join(b)
```

**Comment by jenshnielsen** [comment]
@astafan8 I think this should be ok. Can you have another look

**Review by astafan8 — APPROVED** [review]
looks good!

do we still need the filterwarnings?

**Comment by jenshnielsen** [comment]
> do we still need the filterwarnings?

The filterwarnings are added since xarray and hdf5netcdf raises these warnings too. I have verified both by running the tests and grepping that we do use distutils internally (other than versioneer)

**Comment by jenshnielsen** [comment]
bors merge

**Pull request merged**